Add some docs for the SQL injection detection (#163)
dbos-inc/dbos-docs: docs/api-reference/static-analysis.md

Many DBOS-suggested coding practices can be enforced by a combination of `eslint` plugins and rule configurations.

### Installing and configuring `eslint`

> ::::tip

Reviewer: We still want to mention that "if you got started with the quickstart, the `@dbos-inc/eslint-plugin` is already installed for you."

Author: Resolved

> To install the `eslint` package and the DBOS plugin:

Reviewer: ```suggestion
To install the DBOS eslint plugin:
```

Author: Done

One custom rule from DBOS, `@dbos-inc/dbos-static-analysis`, is provided in the [`@dbos-inc/eslint-plugin`](https://github.com/dbos-inc/eslint-plugin) package.  This rule is enabled by default.

___

Running a database query with a string that's vulnerable to SQL injection will result in an error.
This would typically happen inside of a [transaction](https://docs.dbos.dev/tutorials/transaction-tutorial).
- SQL injection happens when a bad actor puts SQL code as a field into something like an online form,
and if a programmer builds a raw query from SQL and this data, the bad actor's supposed data may allow them to run
arbitrary SQL commands over your database.
- To avoid injection, use parameterized queries. But if you accidentally make yourself vulnerable to injection, DBOS is here to save you!

Here's how you should make a parameterized query:
```ts
export class Greetings {
  @Transaction()
  static async InsertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
    await ctxt.client.raw('INSERT INTO greetings (name, note) VALUES (?, ?)', [friend, note]);
  }
}
```

> Here's how you would make SQL injection possible:

Reviewer: ```suggestion
This example is vulnerable to SQL injection:
```

```ts
export class Greetings {
  @Transaction()

>   static async InsertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {

Reviewer: ```suggestion
  static async VulnerableGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
```

>     await ctxt.client.raw(`INSERT INTO greetings (name, note) VALUES (${friend}, ${note})`);

Reviewer: ```suggestion
    await ctxt.client.raw(`INSERT INTO greetings (name, note) VALUES (${friend}, ${note})`); // Don't do this!
```

  }
}
```

- The plugin will raise a potential SQL injection error if your query string is either directly or indirectly built up of a nonliteral component.

> - For example, if your query is a format string that has formatting parameters of things like a function call, a parameter, or something else like that, the plugin will let you know that you're vulnerable to injection.

Reviewer: ```suggestion
- For example, if your query is a format string parameterized by a function call, input variable, or other non-literal inputs, the plugin will let you know that you're vulnerable to injection.
```

One custom rule from DBOS, `@dbos-inc/dbos-static-analysis`, is provided in the [`@dbos-inc/eslint-plugin`](https://github.com/dbos-inc/eslint-plugin) package.  This rule is enabled by default.

___

Running a database query with a string that's vulnerable to SQL injection will result in an error.
This would typically happen inside of a [transaction](https://docs.dbos.dev/tutorials/transaction-tutorial).
- SQL injection happens when a bad actor puts SQL code as a field into something like an online form,
and if a programmer builds a raw query from SQL and this data, the bad actor's supposed data may allow them to run
arbitrary SQL commands over your database.

> - To avoid injection, use prepared statements. But if you accidentally make yourself vulnerable to injection, DBOS is here to save you!

Reviewer: Replace prepared statement -> parameterized query

Author: Done

Here's how you should make a prepared statement:
```ts
export class Greetings {
  @Transaction()
  static async InsertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
    await ctxt.client.raw('INSERT INTO greetings (name, note) VALUES (?, ?)', [friend, note]);
  }
}
```

Here's how you would make SQL injection possible:
```ts
export class Greetings {
  @Transaction()
  static async InsertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
    await ctxt.client.raw(`INSERT INTO greetings (name, note) VALUES (${friend}, ${note})`);
  }
}
```

> - A LR-value is a literal-reducible value, or a value that is known to be literal at some level.

Reviewer: Let's remove this paragraph as it's an implementation detail. Replace it with plain language instead.

Author: Done
